core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -2263,37 +2271,43 @@ class KafkaController(val config: KafkaConfig,
             .groupBy { case (tp, _) => tp.topic }   // Group by topic
             .foreach { case (topic, partitions) =>
               // Add each topic part to the response
-              val topicResp = new AlterIsrResponseData.TopicData()
+              val topicResp = new AlterPartitionResponseData.TopicData()
                 .setName(topic)
                 .setPartitions(new util.ArrayList())
               resp.topics.add(topicResp)
               partitions.foreach { case (tp, errorOrIsr) =>
                 // Add each partition part to the response (new ISR or error)
                 errorOrIsr match {
                   case Left(error) => topicResp.partitions.add(
-                    new AlterIsrResponseData.PartitionData()
+                    new AlterPartitionResponseData.PartitionData()
                       .setPartitionIndex(tp.partition)
                       .setErrorCode(error.code))
                   case Right(leaderAndIsr) => topicResp.partitions.add(
-                    new AlterIsrResponseData.PartitionData()
+                    new AlterPartitionResponseData.PartitionData()
                       .setPartitionIndex(tp.partition)
                       .setLeaderId(leaderAndIsr.leader)
                       .setLeaderEpoch(leaderAndIsr.leaderEpoch)
                       .setIsr(leaderAndIsr.isr.map(Integer.valueOf).asJava)
-                      .setCurrentIsrVersion(leaderAndIsr.zkVersion))
+                      .setLeaderRecoveryState(leaderAndIsr.leaderRecoveryState.value)
```

Comment:
It maybe a little subtle but I don't think we need to. If the version is `0` then this is guaranteed to be the default value `0` so the serialization will succeed. This is true because we only write these values in the response when the operation success. If the operation fails then we skip writing these values and instead just write the error code.

Reply:
Based on this discussion https://github.com/apache/kafka/pull/11733#issuecomment-1051066928, I decided to mark this field as ignorable in the response.